### ISSUE TITLE
Allow overriding download url (which helps in the case of bad download locations)

### ISSUE THREAD

**Issue by ferventcoder** [chocolatey/choco#467]
This would make @rismoney so happy!

**Comment by ferventcoder** [comment]
I looked over in chocolatey/chocolatey to find the old issue that mentioned this, but maybe it was just in email threads and never here.

**Comment by rismoney** [comment]
> This would make @rismoney so happy!

The particular use case is this-

All my package binaries are currently fetching from an internal repo say:  chocopackages.rismoney.com
This makes all those packages non-portable.  I would like to put in the URL a token, and have chocolatey insert the hostname location.

this way i could code all my packages w/o using chocopackages.rismoney.com and instead use the token.

This would go a long way towards open sourcing our packages, and have people be able to use them on their internal repos w/o modifying the chocolateyinstall.ps1 for their org.

**Comment by gep13** [comment]
Rismoney, wouldn't you be able to do what you are currently describing using the PackageParameters input to the choco install/upgrade commands?

Sent from my Windows Phone
________________________________

**Comment by rismoney** [comment]
yes, but the key is i don't want to specify it on the command line.  i imagine it could be contained within the xml config file.

**Comment by i-give-up** [comment]
Can I use this with Chocolatey's feed? I'm having trouble with Fosshub links.

And yes, being able to set this in a config file would be nice.

**Comment by gep13** [comment]
>@rismoney said...
but the key is i don't want to specify it on the command line.

Ah, I see what you are getting at.  I like that idea, but I would see that as a separate issue to this one, which allows overriding of the URL.  Great idea!

**Comment by gep13** [comment]
@i-give-up this is just an issue to track the work that needs to be done, and hasn't yet been implemented, but yes, the idea would be that you could use it on any package, regardless of where it is hosted.

**Comment by ferventcoder** [comment]
@rismoney @gep13 yes, that is more about something businesses would take advantage of. :)

**Comment by ferventcoder** [comment]
This is going to be a trickle down feature, first available in pro+ editions and later available in open source.
